octane-eclipse-plugin/src/com/hpe/octane/ideplugins/eclipse/ui/entitydetail/field/FieldEditorFactory.java

```diff
@@ -85,4 +126,67 @@ public FieldEditor createFieldEditor(Composite parent, EntityModelWrapper entity
         return fieldEditor;
     }
 
-}
+    private static void disposeFieldEditor(FieldEditor fieldEditor) {
+        if (fieldEditor != null && fieldEditor instanceof Control) {
+            ((Control) fieldEditor).dispose();
+        }
+    }
+
+    private FieldEditor createReferenceFieldEditor(Composite parent, EntityModelWrapper entityModelWrapper, FieldMetadata fieldMetadata) {
+
+        Target[] targets = fieldMetadata.getFieldTypedata().getTargets();
+        if (targets.length != 1) {
+            throw new RuntimeException("Multiple target refrence fields not supported");
+        }
+
+        ReferenceFieldEditor fieldEditor = new ReferenceFieldEditor(parent, SWT.NONE);
+
+        if (fieldMetadata.getFieldTypedata().isMultiple()) {
+            fieldEditor.setSelectionMode(SWT.MULTI);
+        } else {
+            fieldEditor.setSelectionMode(SWT.SINGLE);
+        }
+
+        Target traget = targets[0];
+        String logicalName = traget.logicalName();
+
+        // List node loader
+        if (Entity.LIST_NODE.getEntityName().equals(traget.getType())) {
+            fieldEditor.setEntityLoader((searchQuery) -> {
+                QueryBuilder qb = Query.statement("list_root", QueryMethod.EqualTo,
+                        Query.statement("logical_name", QueryMethod.EqualTo, logicalName));
+                                
+                Collection<EntityModel> entities = entityService.findEntities(Entity.LIST_NODE, qb, null);
+                
+                //for some reason list nodes are not server side filterable, so you have to do it client side   
+                if(!searchQuery.isEmpty()) {
+                    entities =
+                        entities
+                        .stream()
+                        .filter(entityModel -> stringLike(Util.getUiDataFromModel(entityModel.getValue(EntityFieldsConstants.FIELD_NAME)), searchQuery))
+                        .collect(Collectors.toList());
+                }
+                
+                return entities;
+            });
+        } else {
+            disposeFieldEditor(fieldEditor);

```

Comment:
@atothhpe Is there a point in instantiating a new fieldEditor if the target type != LIST_NODE? Perhaps it can be refactored:
if (target type not supported)
  throw
else {
  instantiate fieldEditor
  fieldEditor.setSelectionMode
  fieldEditor.setEntityLoaded...
}

Reply:
that's a great idea, I forgot I could just save the loader in a local variable, (dum dum)